components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/dao/impl/ActionManagementDAOImpl.java

```diff
@@ -224,6 +256,81 @@ public Action getActionByActionId(String actionId, Integer tenantId) throws Acti
         }
     }
 
+    @Override
+    public Action updateActionEndpointAuthProperties(String actionId, AuthType authentication, int tenantId)
+            throws ActionMgtException {
+
+        Connection dbConnection = IdentityDatabaseUtil.getDBConnection(true);
+        try {
+            Map<String, String> nonSecretEndpointProperties = authentication.getProperties().stream()
+                    .filter(property -> !property.getIsConfidential())
+                    .collect(Collectors.toMap(AuthProperty::getName, AuthProperty::getValue));
+            // Update non-secret endpoint properties.
+            updateActionEndpointProperties(dbConnection, actionId, nonSecretEndpointProperties, tenantId);
+            // Encrypt and update non-secret endpoint properties.
+            actionSecretProcessor.encryptAssociatedSecrets(authentication.getProperties(), actionId,
+                    authentication.getType().name());
```

Comment:
This function seems to be returning all properties while having non secrets not encrypted. So do you want to update non secret properties separately ?

If you want to split, split it consistently

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5795/commits/4c40292ff2e65531731f8e68897c74215dae5279